src/datetime-picker.component.ts

```diff
@@ -140,6 +140,15 @@ declare var moment: any;
   `,
   styles: [
     `
+* {
```

Comment:
this is questoinable. 
Why don't we make our css to fit to  box-sizing to border-box and set it to box-sizing to border-box.
The reason is that the css is overridable by app or index.html, and others may not want to change  box-sizing to content-box because border-box is widely recommended once used.